### ISSUE TITLE
Wrong semicolon hints for JS files

### ISSUE THREAD

**Issue by qmegas** [apache/netbeans#4214]
### Apache NetBeans version

Apache NetBeans 13

### What happened

There are 2 small ES6 paring errors. I decided to report them in a single ticket, but can split if required.

**Probem 1**
Example code:

```js
class A {
}
class B extends A {
}
```

When extending class there is incorrect warning "_Expected semicolon ; after }_" on line 4

 
**Problem 2**
Example code:

```js
const a = {
	foo: 1,
	bar: 2,
};
```
On line 3 there is incorrect warning "_Unexpected ','_" however the syntax is correct.

### Did this work correctly in an earlier version?

No

### Operating System

Windows 10

### JDK

16.0.1

### Apache NetBeans packaging

Apache NetBeans provided installer

### Anything else

That issue was reported by me 2 years ago https://issues.apache.org/jira/browse/NETBEANS-4584

### Are you willing to submit a pull request?

No

### Code of Conduct

Yes

**Comment by matthiasblaesing** [comment]
To be clear: These are not parsing errors, these are errors in the hints module. The parse happily returns the AST for these constructs, it is just the analysis, that fails to properly handle these cases.

**Comment by matthiasblaesing** [comment]
Ok - here is only one bug. Problem 2 is just a misconfiguration - just deactivate the hint:

![image](https://user-images.githubusercontent.com/2179736/174154603-34016532-a123-4d7f-9627-1d08a3740384.png)

Problem 1 should be solved in the referenced PR #4232 and there is also a test build linked. Please test it.
